packages/expo-asset/src/Asset.ts

```diff
@@ -18,7 +18,7 @@ type AssetDescriptor = {
 };
 
 type DownloadPromiseCallbacks = {
-  resolve: () => void;
+  resolve: (value?: any) => void;
```

Comment:
resolve typing was updated: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#resolves-parameters-are-no-longer-optional-in-promises

Reply:
From that link it sounds like we should make clear that promise is `Promise<void>` instead of changing the signature here.